src/main/java/ir/sahab/dropwizardmetrics/LabeledMetric.java

```diff
@@ -54,6 +54,16 @@ public static boolean hasLabel(String nameAndLabels) {
                 && nameAndLabels.indexOf(']') == nameAndLabels.lastIndexOf(']');
     }
 
+    /**
+     * Extracts the name from a labeled metric name.
+     */
+    public static String extractMetricName(String labeledMetricName) {
+        if (!hasLabel(labeledMetricName)) {
+            return labeledMetricName;
+        }
+        return labeledMetricName.substring(0, labeledMetricName.indexOf("["));
+    }
+
```

Comment:
Use 'labeledMetricName' consistently for the argument of other methods too:

hasLabel(String labeledMetricName)
private constructor: LabeledMetric(String labeledMetricName)
processLabels(String labeledMetricName, Consumer...)

Reply:
Done